lib/web_ui/lib/src/engine/canvaskit/canvaskit_api.dart

```diff
@@ -2666,40 +2667,83 @@ void patchCanvasKitModule(DomHTMLScriptElement canvasKitScript) {
   domDocument.head!.appendChild(canvasKitScript);
 }
 
-String get canvasKitBuildUrl => configuration.canvasKitBaseUrl;
-String get canvasKitJavaScriptBindingsUrl =>
-    '${canvasKitBuildUrl}canvaskit.js';
-String canvasKitWasmModuleUrl(String canvasKitBase, String file) =>
+const String _kFullCanvasKitJsFileName = 'canvaskit.js';
+const String _kChromiumCanvasKitJsFileName = 'chromium/canvaskit.js';
+
+String get _canvasKitBaseUrl => configuration.canvasKitBaseUrl;
+List<String> get _canvasKitJsFileNames {
+  switch (configuration.canvasKitVariant) {
+    case CanvasKitVariant.auto:
+      return <String>[
+        if (browserSupportsCanvaskitChromium) _kChromiumCanvasKitJsFileName,
+        _kFullCanvasKitJsFileName,
+      ];
+    case CanvasKitVariant.full:
+      return <String>[_kFullCanvasKitJsFileName];
+    case CanvasKitVariant.chromium:
+      return <String>[_kChromiumCanvasKitJsFileName];
+  }
```

Comment:
I think this mode is allowing people to configure their apps in a way that is quite breaking to part of their potential users.

I agree that we should offer "auto" (try to optimize) and "full" (opt out of the optimization), but "chromium" mode sounds like trouble down the line.

(Unless we want to bring back the "Best Viewed With Internet Explorer" type of badges :P)

Reply:
My two cents, I think we should allow it, just for clarity and symmetry. If people really shoot themselves in the foot with this and make their app unusable outside of Chromium browsers, that's on them. This might be useful in some sort of testing edge cases or something.

Reply:
> If people really shoot themselves in the foot with this and make their app unusable outside of Chromium browsers, that's on them.

I don't know, it's possible I might be jaded by the discussions I have with people on the internet about CORS and other basic things :/

In any case, "chromium" mode smells like YAGNI. If we add it for testing or other edge cases, I'd call it: "dangerously-set-chromium-breaks-other-browsers" :P

Reply:
Good point. The only use case I could think of is testing.

For testing, we may want to force the chromium variant. If we leave it at auto, and we forget to copy canvaskit_chromium, then the tests will silently fallback to the "full" canvaskit and we won't notice.

I'm happy to add some warnings in the dartdocs. We could also change the string to "unsafe_chromium" but I'm hesitant to do it because I don't want people to think that using the new lite canvaskit is unsafe.

Reply:
OK, I think that if it's only for testing we should be able to override this differently from the engine build, but let's add it and see what happens :P